bin/makePsfLibrary.py

```diff
@@ -0,0 +1,181 @@
+    # needed to remove the optical distortion from the Psf. Use R22_S11 as center
+    butler = dafPersist.Butler(root=os.path.join(os.path.dirname(__file__), "."))
+    camera = butler.get("camera", immediate=True)
+    det = camera["R:2,2 S:1,1"]
+    tm = det.getTransformMap()
+    for sys in tm.getCoordSysList():
+        if sys.getSysName() == "Pixels":
+            transform1 = tm.get(sys).invert()
+        if sys.getSysName() == "TanPixels":
+            transform2 = tm.get(sys)
+    xytransform = afwGeom.MultiXYTransform([transform1, transform2])
+    
+    # A centroid algorithm has to be run here, because the catalog from OpSim
+    # is not very accurate
+    schema = afwTable.SourceTable.makeMinimalSchema()
+    control = measBase.GaussianCentroidControl()
+    alg = measBase.GaussianCentroidAlgorithm
+    plugin = alg(control, "", schema)
+    
+    # count of psfs actually created
+    psfIndex = 1
+    # Go through all the images and write out the galaxies
+    psfCount = 0
+    outputHdus = pyfits.hdu.hdulist.HDUList()
+    for image in images:
+        exp = afwImage.ExposureF(image)
+        cat = open(catname, "r")
+        # number the psfs the same way they are listed in the catalog, starting with line 1
+        # read through the OpSim catalog and create a psf for any object which is completely
+        # inside the image we are processing
+        while True:
+            line = cat.readline()
+            if len(line) == 0:
+                break
+            # catalog lines with objects have x,y position at index 2,3.  Use that for 1st guess
+            attrs = line.split()
+            if attrs[0] != "object": continue
+            coord = afwCoord.Coord(afwGeom.Point2D(float(attrs[2]), float(attrs[3])))
+            pixels = exp.getWcs().skyToPixel(coord)
+            x = int(pixels.getX())
+            y = int(pixels.getY())
+            starBBox = afwGeom.Box2I(afwGeom.Point2I(x, y), afwGeom.Extent2I(1,1))
+            starBBox.grow(psfSize)
+        
+            # test to be sure that the star is completely contained
+            if exp.getBBox().contains(starBBox):
+                subImage = afwImage.ImageF(exp.getMaskedImage().getImage(), starBBox)
+                ds = afwDetection.FootprintSet(subImage, afwDetection.Threshold(1000))
+                sources = afwTable.SourceCatalog(schema)
+                record = sources.addNew()
+                record.setFootprint(ds.getFootprints()[0])
+                plugin.measure(record, exp)
+        
+                # shift the bounding box to the correct centroid
+                newCoord = exp.getWcs().pixelToSky(afwGeom.Point2D(record.get("_x"), record.get("_y")))
+                dx = int(round(record.get("_x"))) - x
+                dy = int(round(record.get("_y"))) - y
+                starBBox.shift(afwGeom.Extent2I(dx, dy))
+        
+                # if the exposure still contains the cutout, use it.
+                if exp.getBBox().contains(starBBox):
```

Comment:
I think this block and the similar `if` statement above it would read slightly better if you instead changed them to:
```
if not exp.getBBox().contains(starBBox):
    continue
```
That would let you remove the indentation, which in this case is a little misleading.